modules/ding_campaign_plus/modules/ding_campaign_plus_basic/js/ding_campaign_plus_basic.admin.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ * Handle rules UI in the administration interface.
+ */
+(function ($) {
+  'use strict';
+
+  // The select type that have auto-complete callbacks.
+  // @TODO: Get these from the backend via drupal.settings.
+  var auto_complete_types = [
+    'page',
+    'event',
+    'news',
+    'library',
+    'group',
+    'taxonomy'
+  ];
+
+  /**
+   * Rebuild auto-complete callbacks when rule type is changed.
+   *
+   * @param $context
+   *   The context the auto-complete is located in.
+   * @param value
+   *   The selected value in the rule type select.
+   */
+  function rebuild_autocomplete($context, value) {
+    var $obj = $('input.form-text', $context);
+    $obj.unbind().removeClass('form-autocomplete').addClass('autocomplete-processed');
```

Comment:
`unbind` has been deprecated since jQuery 1.7 (cf. http://api.jquery.com/unbind/#unbind). Unless we rely on a very old version of jQuery, please use `off`.

Reply:
Changed